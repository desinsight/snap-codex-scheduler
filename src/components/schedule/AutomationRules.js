import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import { fetchAutomationRules, createAutomationRule, deleteAutomationRule, toggleAutomationRule, } from '../../store/slices/automationSlice';
const Container = styled.div `
  padding: 20px;
`;
const Title = styled.h2 `
  margin-bottom: 20px;
`;
const RuleList = styled.div `
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  gap: 20px;
  margin-bottom: 20px;
`;
const RuleCard = styled.div `
  background: #f5f5f5;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 15px;
`;
const RuleHeader = styled.div `
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
`;
const RuleName = styled.h3 `
  margin: 0;
`;
const RuleType = styled.span `
  font-size: 0.9em;
  color: #666;
`;
const RuleConditions = styled.div `
  margin-bottom: 10px;
`;
const RuleActions = styled.div `
  margin-bottom: 10px;
`;
const ToggleButton = styled.button `
  background: ${props => (props.active ? '#4CAF50' : '#f44336')};
  color: white;
  border: none;
  padding: 5px 10px;
  border-radius: 4px;
  cursor: pointer;
`;
const Form = styled.form `
  background: #f5f5f5;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 20px;
  margin-top: 20px;
`;
const FormGroup = styled.div `
  margin-bottom: 15px;
`;
const Label = styled.label `
  display: block;
  margin-bottom: 5px;
`;
const Input = styled.input `
  width: 100%;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
`;
const Select = styled.select `
  width: 100%;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
`;
const Button = styled.button `
  background: #2196f3;
  color: white;
  border: none;
  padding: 10px 20px;
  border-radius: 4px;
  cursor: pointer;
`;
const AutomationRules = () => {
    const { t } = useTranslation('schedule');
    const dispatch = useDispatch();
    const { rules, loading, error } = useSelector((state) => state.automation);
    const [newRule, setNewRule] = useState({
        name: '',
        type: 'channelSwitch',
        conditions: [],
        actions: [],
        isActive: true,
    });
    useEffect(() => {
        dispatch(fetchAutomationRules());
    }, [dispatch]);
    const handleInputChange = (e) => {
        const { name, value } = e.target;
        setNewRule(prev => ({ ...prev, [name]: value }));
    };
    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(createAutomationRule(newRule));
        setNewRule({
            name: '',
            type: 'channelSwitch',
            conditions: [],
            actions: [],
            isActive: true,
        });
    };
    const handleToggle = (ruleId) => {
        dispatch(toggleAutomationRule(ruleId));
    };
    const handleDelete = (ruleId) => {
        if (window.confirm(t('notifications.automation.confirmDelete'))) {
            dispatch(deleteAutomationRule(ruleId));
        }
    };
    if (loading) {
        return _jsx("div", { children: t('common.loading') });
    }
    if (error) {
        return _jsx("div", { children: error });
    }
    return (_jsxs(Container, { children: [_jsx(Title, { children: t('notifications.automation.title') }), _jsx(RuleList, { children: rules.map((rule) => (_jsxs(RuleCard, { children: [_jsxs(RuleHeader, { children: [_jsx(RuleName, { children: rule.name }), _jsx(ToggleButton, { active: rule.isActive, onClick: () => handleToggle(rule.id), children: rule.isActive
                                        ? t('notifications.automation.deactivate')
                                        : t('notifications.automation.activate') })] }), _jsx(RuleType, { children: t(`notifications.automation.types.${rule.type}`) }), _jsxs(RuleConditions, { children: [_jsx("h4", { children: t('notifications.automation.conditions') }), rule.conditions.map((condition, index) => (_jsxs("div", { children: [condition.metric, " ", condition.operator, " ", condition.value] }, index)))] }), _jsxs(RuleActions, { children: [_jsx("h4", { children: t('notifications.automation.actions') }), rule.actions.map((action, index) => (_jsxs("div", { children: [action.type, ": ", JSON.stringify(action.value)] }, index)))] }), _jsx(Button, { onClick: () => handleDelete(rule.id), children: t('notifications.automation.delete') })] }, rule.id))) }), _jsxs(Form, { onSubmit: handleSubmit, children: [_jsx("h3", { children: t('notifications.automation.createRule') }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.automation.name') }), _jsx(Input, { type: "text", name: "name", value: newRule.name, onChange: handleInputChange, required: true })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.automation.type') }), _jsxs(Select, { name: "type", value: newRule.type, onChange: handleInputChange, required: true, children: [_jsx("option", { value: "channelSwitch", children: t('notifications.automation.types.channelSwitch') }), _jsx("option", { value: "priorityAdjustment", children: t('notifications.automation.types.priorityAdjustment') }), _jsx("option", { value: "retryStrategy", children: t('notifications.automation.types.retryStrategy') }), _jsx("option", { value: "fatigueManagement", children: t('notifications.automation.types.fatigueManagement') })] })] }), _jsx(Button, { type: "submit", children: t('notifications.automation.create') })] })] }));
};
export default AutomationRules;
