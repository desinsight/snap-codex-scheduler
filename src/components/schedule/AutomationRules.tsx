import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import {
  fetchAutomationRules,
  createAutomationRule,
  updateAutomationRule,
  deleteAutomationRule,
  toggleAutomationRule,
} from '../../store/slices/automationSlice';
import { AutomationRule } from '../../types/notification';

const Container = styled.div`
  padding: 20px;
`;

const Title = styled.h2`
  margin-bottom: 20px;
`;

const RuleList = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  gap: 20px;
  margin-bottom: 20px;
`;

const RuleCard = styled.div`
  background: #f5f5f5;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 15px;
`;

const RuleHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
`;

const RuleName = styled.h3`
  margin: 0;
`;

const RuleType = styled.span`
  font-size: 0.9em;
  color: #666;
`;

const RuleConditions = styled.div`
  margin-bottom: 10px;
`;

const RuleActions = styled.div`
  margin-bottom: 10px;
`;

const ToggleButton = styled.button<{ active: boolean }>`
  background: ${(props) => (props.active ? '#4CAF50' : '#f44336')};
  color: white;
  border: none;
  padding: 5px 10px;
  border-radius: 4px;
  cursor: pointer;
`;

const Form = styled.form`
  background: #f5f5f5;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 20px;
  margin-top: 20px;
`;

const FormGroup = styled.div`
  margin-bottom: 15px;
`;

const Label = styled.label`
  display: block;
  margin-bottom: 5px;
`;

const Input = styled.input`
  width: 100%;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
`;

const Select = styled.select`
  width: 100%;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
`;

const Button = styled.button`
  background: #2196F3;
  color: white;
  border: none;
  padding: 10px 20px;
  border-radius: 4px;
  cursor: pointer;
`;

const AutomationRules: React.FC = () => {
  const { t } = useTranslation('schedule');
  const dispatch = useDispatch();
  const { rules, loading, error } = useSelector((state: any) => state.automation);
  const [newRule, setNewRule] = useState<Partial<AutomationRule>>({
    name: '',
    type: 'channelSwitch',
    conditions: [],
    actions: [],
    isActive: true,
  });

  useEffect(() => {
    dispatch(fetchAutomationRules());
  }, [dispatch]);

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
    const { name, value } = e.target;
    setNewRule((prev) => ({ ...prev, [name]: value }));
  };

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    dispatch(createAutomationRule(newRule as Omit<AutomationRule, 'id' | 'createdAt' | 'updatedAt'>));
    setNewRule({
      name: '',
      type: 'channelSwitch',
      conditions: [],
      actions: [],
      isActive: true,
    });
  };

  const handleToggle = (ruleId: string) => {
    dispatch(toggleAutomationRule(ruleId));
  };

  const handleDelete = (ruleId: string) => {
    if (window.confirm(t('notifications.automation.confirmDelete'))) {
      dispatch(deleteAutomationRule(ruleId));
    }
  };

  if (loading) {
    return <div>{t('common.loading')}</div>;
  }

  if (error) {
    return <div>{error}</div>;
  }

  return (
    <Container>
      <Title>{t('notifications.automation.title')}</Title>
      <RuleList>
        {rules.map((rule: AutomationRule) => (
          <RuleCard key={rule.id}>
            <RuleHeader>
              <RuleName>{rule.name}</RuleName>
              <ToggleButton
                active={rule.isActive}
                onClick={() => handleToggle(rule.id)}
              >
                {rule.isActive ? t('notifications.automation.deactivate') : t('notifications.automation.activate')}
              </ToggleButton>
            </RuleHeader>
            <RuleType>{t(`notifications.automation.types.${rule.type}`)}</RuleType>
            <RuleConditions>
              <h4>{t('notifications.automation.conditions')}</h4>
              {rule.conditions.map((condition, index) => (
                <div key={index}>
                  {condition.metric} {condition.operator} {condition.value}
                </div>
              ))}
            </RuleConditions>
            <RuleActions>
              <h4>{t('notifications.automation.actions')}</h4>
              {rule.actions.map((action, index) => (
                <div key={index}>
                  {action.type}: {JSON.stringify(action.value)}
                </div>
              ))}
            </RuleActions>
            <Button onClick={() => handleDelete(rule.id)}>
              {t('notifications.automation.delete')}
            </Button>
          </RuleCard>
        ))}
      </RuleList>
      <Form onSubmit={handleSubmit}>
        <h3>{t('notifications.automation.createRule')}</h3>
        <FormGroup>
          <Label>{t('notifications.automation.name')}</Label>
          <Input
            type="text"
            name="name"
            value={newRule.name}
            onChange={handleInputChange}
            required
          />
        </FormGroup>
        <FormGroup>
          <Label>{t('notifications.automation.type')}</Label>
          <Select
            name="type"
            value={newRule.type}
            onChange={handleInputChange}
            required
          >
            <option value="channelSwitch">{t('notifications.automation.types.channelSwitch')}</option>
            <option value="priorityAdjustment">{t('notifications.automation.types.priorityAdjustment')}</option>
            <option value="retryStrategy">{t('notifications.automation.types.retryStrategy')}</option>
            <option value="fatigueManagement">{t('notifications.automation.types.fatigueManagement')}</option>
          </Select>
        </FormGroup>
        <Button type="submit">{t('notifications.automation.create')}</Button>
      </Form>
    </Container>
  );
};

export default AutomationRules; 