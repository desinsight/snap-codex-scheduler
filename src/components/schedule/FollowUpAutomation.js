import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import { fetchFollowUpRules, createFollowUpRule, deleteFollowUpRule, toggleFollowUpRule, } from '../../store/slices/followUpAutomationSlice';
const Container = styled.div `
  padding: 20px;
`;
const Title = styled.h2 `
  margin-bottom: 20px;
`;
const RuleList = styled.div `
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  gap: 20px;
  margin-bottom: 20px;
`;
const RuleCard = styled.div `
  background: ${({ active }) => (active ? '#f0f8ff' : '#f5f5f5')};
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 15px;
  position: relative;
`;
const RuleHeader = styled.div `
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
`;
const RuleName = styled.h3 `
  margin: 0;
`;
const ToggleButton = styled.button `
  background: ${({ active }) => (active ? '#4caf50' : '#f44336')};
  color: white;
  border: none;
  border-radius: 4px;
  padding: 5px 10px;
  cursor: pointer;
`;
const RuleContent = styled.div `
  margin-bottom: 10px;
`;
const RuleLabel = styled.div `
  font-weight: bold;
  margin-bottom: 5px;
`;
const RuleValue = styled.div `
  margin-bottom: 10px;
`;
const ActionButtons = styled.div `
  display: flex;
  gap: 10px;
  margin-top: 10px;
`;
const Button = styled.button `
  background: #2196f3;
  color: white;
  border: none;
  border-radius: 4px;
  padding: 5px 10px;
  cursor: pointer;

  &:hover {
    background: #1976d2;
  }
`;
const DeleteButton = styled(Button) `
  background: #f44336;

  &:hover {
    background: #d32f2f;
  }
`;
const Form = styled.form `
  background: #f5f5f5;
  border: 1px solid #ddd;
  border-radius: 8px;
  padding: 20px;
  margin-top: 20px;
`;
const FormGroup = styled.div `
  margin-bottom: 15px;
`;
const Label = styled.label `
  display: block;
  margin-bottom: 5px;
  font-weight: bold;
`;
const Input = styled.input `
  width: 100%;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
`;
const Select = styled.select `
  width: 100%;
  padding: 8px;
  border: 1px solid #ddd;
  border-radius: 4px;
`;
const FollowUpAutomation = () => {
    const { t } = useTranslation('schedule');
    const dispatch = useDispatch();
    const { rules, loading, error } = useSelector((state) => state.followUpAutomation);
    const [newRule, setNewRule] = useState({
        name: '',
        conditions: {
            responseType: 'ignore',
            timeSinceNotification: 24,
            priority: 'medium',
        },
        actions: {
            channel: 'email',
            templateId: '',
            delay: 1,
            maxRetries: 3,
        },
        isActive: true,
    });
    useEffect(() => {
        dispatch(fetchFollowUpRules());
    }, [dispatch]);
    const handleInputChange = (e) => {
        const { name, value } = e.target;
        if (name.startsWith('conditions.')) {
            const field = name.split('.')[1];
            setNewRule({
                ...newRule,
                conditions: {
                    ...newRule.conditions,
                    [field]: value,
                },
            });
        }
        else if (name.startsWith('actions.')) {
            const field = name.split('.')[1];
            setNewRule({
                ...newRule,
                actions: {
                    ...newRule.actions,
                    [field]: value,
                },
            });
        }
        else {
            setNewRule({
                ...newRule,
                [name]: value,
            });
        }
    };
    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(createFollowUpRule(newRule));
        setNewRule({
            name: '',
            conditions: {
                responseType: 'ignore',
                timeSinceNotification: 24,
                priority: 'medium',
            },
            actions: {
                channel: 'email',
                templateId: '',
                delay: 1,
                maxRetries: 3,
            },
            isActive: true,
        });
    };
    const handleToggle = (ruleId) => {
        dispatch(toggleFollowUpRule(ruleId));
    };
    const handleDelete = (ruleId) => {
        if (window.confirm(t('notifications.followUp.confirmDelete'))) {
            dispatch(deleteFollowUpRule(ruleId));
        }
    };
    if (loading) {
        return _jsx("div", { children: t('common.loading') });
    }
    if (error) {
        return _jsx("div", { children: error });
    }
    return (_jsxs(Container, { children: [_jsx(Title, { children: t('notifications.followUp.title') }), _jsx(RuleList, { children: rules.map((rule) => (_jsxs(RuleCard, { active: rule.isActive, children: [_jsxs(RuleHeader, { children: [_jsx(RuleName, { children: rule.name }), _jsx(ToggleButton, { active: rule.isActive, onClick: () => handleToggle(rule.id), children: rule.isActive ? t('common.active') : t('common.inactive') })] }), _jsxs(RuleContent, { children: [_jsx(RuleLabel, { children: t('notifications.followUp.conditions') }), _jsxs(RuleValue, { children: [t('notifications.followUp.responseType'), ": ", rule.conditions.responseType, _jsx("br", {}), t('notifications.followUp.timeSinceNotification'), ":", ' ', rule.conditions.timeSinceNotification, "h", _jsx("br", {}), t('notifications.followUp.priority'), ": ", rule.conditions.priority] }), _jsx(RuleLabel, { children: t('notifications.followUp.actions') }), _jsxs(RuleValue, { children: [t('notifications.followUp.channel'), ": ", rule.actions.channel, _jsx("br", {}), t('notifications.followUp.delay'), ": ", rule.actions.delay, "h", _jsx("br", {}), t('notifications.followUp.maxRetries'), ": ", rule.actions.maxRetries] })] }), _jsxs(ActionButtons, { children: [_jsx(Button, { onClick: () => handleToggle(rule.id), children: rule.isActive ? t('common.deactivate') : t('common.activate') }), _jsx(DeleteButton, { onClick: () => handleDelete(rule.id), children: t('common.delete') })] })] }, rule.id))) }), _jsxs(Form, { onSubmit: handleSubmit, children: [_jsx("h3", { children: t('notifications.followUp.createRule') }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.ruleName') }), _jsx(Input, { type: "text", name: "name", value: newRule.name, onChange: handleInputChange, required: true })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.responseType') }), _jsxs(Select, { name: "conditions.responseType", value: newRule.conditions.responseType, onChange: handleInputChange, children: [_jsx("option", { value: "ignore", children: t('notifications.followUp.ignore') }), _jsx("option", { value: "dismiss", children: t('notifications.followUp.dismiss') })] })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.timeSinceNotification') }), _jsx(Input, { type: "number", name: "conditions.timeSinceNotification", value: newRule.conditions.timeSinceNotification, onChange: handleInputChange, min: "1", required: true })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.priority') }), _jsxs(Select, { name: "conditions.priority", value: newRule.conditions.priority, onChange: handleInputChange, children: [_jsx("option", { value: "low", children: t('notifications.followUp.low') }), _jsx("option", { value: "medium", children: t('notifications.followUp.medium') }), _jsx("option", { value: "high", children: t('notifications.followUp.high') })] })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.channel') }), _jsx(Select, { name: "actions.channel", value: newRule.actions.channel, onChange: handleInputChange, children: Object.values(NotificationChannel).map(channel => (_jsx("option", { value: channel, children: channel }, channel))) })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.delay') }), _jsx(Input, { type: "number", name: "actions.delay", value: newRule.actions.delay, onChange: handleInputChange, min: "1", required: true })] }), _jsxs(FormGroup, { children: [_jsx(Label, { children: t('notifications.followUp.maxRetries') }), _jsx(Input, { type: "number", name: "actions.maxRetries", value: newRule.actions.maxRetries, onChange: handleInputChange, min: "1", required: true })] }), _jsx(Button, { type: "submit", children: t('notifications.followUp.create') })] })] }));
};
export default FollowUpAutomation;
