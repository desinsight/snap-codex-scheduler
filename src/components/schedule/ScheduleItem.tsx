import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { Schedule, ScheduleCategory } from '../../types/schedule';
import { format } from 'date-fns';

const ScheduleCard = styled.div`
  background-color: white;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  padding: 1rem;
  transition: transform 0.2s;

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  }
`;

const ScheduleHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 0.5rem;
`;

const ScheduleTitle = styled.h3`
  margin: 0;
  font-size: 1.1rem;
  color: #333;
`;

const ScheduleCategoryBadge = styled.span<{ category: ScheduleCategory }>`
  padding: 0.25rem 0.5rem;
  border-radius: 4px;
  font-size: 0.8rem;
  font-weight: 500;
  background-color: ${props => {
    switch (props.category) {
      case ScheduleCategory.WORK:
        return '#e3f2fd';
      case ScheduleCategory.PERSONAL:
        return '#f3e5f5';
      case ScheduleCategory.EDUCATION:
        return '#e8f5e9';
      case ScheduleCategory.HEALTH:
        return '#fff3e0';
      default:
        return '#f5f5f5';
    }
  }};
  color: ${props => {
    switch (props.category) {
      case ScheduleCategory.WORK:
        return '#1976d2';
      case ScheduleCategory.PERSONAL:
        return '#7b1fa2';
      case ScheduleCategory.EDUCATION:
        return '#2e7d32';
      case ScheduleCategory.HEALTH:
        return '#e65100';
      default:
        return '#616161';
    }
  }};
`;

const ScheduleDescription = styled.p`
  margin: 0.5rem 0;
  color: #666;
  font-size: 0.9rem;
`;

const ScheduleTime = styled.div`
  color: #666;
  font-size: 0.9rem;
  margin-bottom: 0.5rem;
`;

const ScheduleFooter = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 1rem;
  padding-top: 0.5rem;
  border-top: 1px solid #eee;
`;

const ScheduleActions = styled.div`
  display: flex;
  gap: 0.5rem;
`;

const ActionButton = styled.button`
  padding: 0.25rem 0.5rem;
  border: none;
  border-radius: 4px;
  background-color: #f5f5f5;
  color: #666;
  font-size: 0.8rem;
  cursor: pointer;
  transition: background-color 0.2s;

  &:hover {
    background-color: #e0e0e0;
  }
`;

const SharedBadge = styled.span`
  padding: 0.25rem 0.5rem;
  border-radius: 4px;
  font-size: 0.8rem;
  background-color: #e3f2fd;
  color: #1976d2;
`;

interface ScheduleItemProps {
  schedule: Schedule;
}

const ScheduleItem: React.FC<ScheduleItemProps> = ({ schedule }) => {
  const { t } = useTranslation();

  const formatDate = (date: Date) => {
    return format(new Date(date), 'MMM d, yyyy HH:mm');
  };

  return (
    <ScheduleCard>
      <ScheduleHeader>
        <ScheduleTitle>{schedule.title}</ScheduleTitle>
        <ScheduleCategoryBadge category={schedule.category}>
          {t(`schedule.category.${schedule.category.toLowerCase()}`)}
        </ScheduleCategoryBadge>
      </ScheduleHeader>

      {schedule.description && (
        <ScheduleDescription>{schedule.description}</ScheduleDescription>
      )}

      <ScheduleTime>
        {schedule.isAllDay
          ? t('schedule.allDay')
          : `${formatDate(schedule.startDate)} - ${formatDate(schedule.endDate)}`}
      </ScheduleTime>

      <ScheduleFooter>
        <ScheduleActions>
          <ActionButton>{t('schedule.actions.edit')}</ActionButton>
          <ActionButton>{t('schedule.actions.delete')}</ActionButton>
        </ScheduleActions>

        {schedule.isShared && (
          <SharedBadge>{t('schedule.shared')}</SharedBadge>
        )}
      </ScheduleFooter>
    </ScheduleCard>
  );
};

export default ScheduleItem; 