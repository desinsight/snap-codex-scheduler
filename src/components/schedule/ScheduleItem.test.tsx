import React from 'react';
import { render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../../../i18n';
import ScheduleItem from './ScheduleItem';
import { Schedule, ScheduleCategory } from '../../types/schedule';

const mockSchedule: Schedule = {
  id: '1',
  title: 'Test Schedule',
  description: 'This is a test schedule',
  startDate: new Date('2024-03-20T10:00:00'),
  endDate: new Date('2024-03-20T12:00:00'),
  isAllDay: false,
  category: ScheduleCategory.WORK,
  isShared: true,
  createdBy: 'user1',
  createdAt: new Date(),
  updatedAt: new Date(),
};

describe('ScheduleItem', () => {
  const renderComponent = (schedule: Schedule) => {
    return render(
      <I18nextProvider i18n={i18n}>
        <ScheduleItem schedule={schedule} />
      </I18nextProvider>
    );
  };

  it('renders schedule title', () => {
    renderComponent(mockSchedule);
    expect(screen.getByText('Test Schedule')).toBeInTheDocument();
  });

  it('renders schedule description', () => {
    renderComponent(mockSchedule);
    expect(screen.getByText('This is a test schedule')).toBeInTheDocument();
  });

  it('renders schedule time correctly', () => {
    renderComponent(mockSchedule);
    expect(screen.getByText(/Mar 20, 2024/)).toBeInTheDocument();
  });

  it('renders category badge', () => {
    renderComponent(mockSchedule);
    expect(screen.getByText('Work')).toBeInTheDocument();
  });

  it('renders shared badge when schedule is shared', () => {
    renderComponent(mockSchedule);
    expect(screen.getByText('Shared')).toBeInTheDocument();
  });

  it('renders action buttons', () => {
    renderComponent(mockSchedule);
    expect(screen.getByText('Edit')).toBeInTheDocument();
    expect(screen.getByText('Delete')).toBeInTheDocument();
  });

  it('renders all-day text when schedule is all day', () => {
    const allDaySchedule = { ...mockSchedule, isAllDay: true };
    renderComponent(allDaySchedule);
    expect(screen.getByText('All Day')).toBeInTheDocument();
  });

  it('does not render description when it is not provided', () => {
    const scheduleWithoutDescription = { ...mockSchedule, description: undefined };
    renderComponent(scheduleWithoutDescription);
    expect(screen.queryByText('This is a test schedule')).not.toBeInTheDocument();
  });

  it('does not render shared badge when schedule is not shared', () => {
    const privateSchedule = { ...mockSchedule, isShared: false };
    renderComponent(privateSchedule);
    expect(screen.queryByText('Shared')).not.toBeInTheDocument();
  });
}); 