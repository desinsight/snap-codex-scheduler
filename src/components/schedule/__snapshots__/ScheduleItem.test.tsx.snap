// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleItem renders schedule title 1`] = `
<div>
  <div
    class="sc-bczRLJ jQJQqE"
  >
    <div
      class="sc-gsnTjR bQJQqE"
    >
      <h3
        class="sc-dkzDqf bQJQqE"
      >
        Test Schedule
      </h3>
      <span
        class="sc-hKMtZM bQJQqE"
        category="WORK"
      >
        Work
      </span>
    </div>
    <p
      class="sc-eCYdqJ bQJQqE"
    >
      This is a test schedule
    </p>
    <div
      class="sc-gKXOVf bQJQqE"
    >
      Mar 20, 2024 10:00 - Mar 20, 2024 12:00
    </div>
    <div
      class="sc-iBkjds bQJQqE"
    >
      <div
        class="sc-jSMfEi bQJQqE"
      >
        <button
          class="sc-furwcr bQJQqE"
        >
          Edit
        </button>
        <button
          class="sc-furwcr bQJQqE"
        >
          Delete
        </button>
      </div>
      <span
        class="sc-jrQzAO bQJQqE"
      >
        Shared
      </span>
    </div>
  </div>
</div>
`;

exports[`ScheduleItem renders all-day text when schedule is all day 1`] = `
<div>
  <div
    class="sc-bczRLJ jQJQqE"
  >
    <div
      class="sc-gsnTjR bQJQqE"
    >
      <h3
        class="sc-dkzDqf bQJQqE"
      >
        Test Schedule
      </h3>
      <span
        class="sc-hKMtZM bQJQqE"
        category="WORK"
      >
        Work
      </span>
    </div>
    <p
      class="sc-eCYdqJ bQJQqE"
    >
      This is a test schedule
    </p>
    <div
      class="sc-gKXOVf bQJQqE"
    >
      All Day
    </div>
    <div
      class="sc-iBkjds bQJQqE"
    >
      <div
        class="sc-jSMfEi bQJQqE"
      >
        <button
          class="sc-furwcr bQJQqE"
        >
          Edit
        </button>
        <button
          class="sc-furwcr bQJQqE"
        >
          Delete
        </button>
      </div>
      <span
        class="sc-jrQzAO bQJQqE"
      >
        Shared
      </span>
    </div>
  </div>
</div>
`;

exports[`ScheduleItem renders category badge 1`] = `
<div>
  <div
    class="sc-bczRLJ jQJQqE"
  >
    <div
      class="sc-gsnTjR bQJQqE"
    >
      <h3
        class="sc-dkzDqf bQJQqE"
      >
        Test Schedule
      </h3>
      <span
        class="sc-hKMtZM bQJQqE"
        category="WORK"
      >
        Work
      </span>
    </div>
    <p
      class="sc-eCYdqJ bQJQqE"
    >
      This is a test schedule
    </p>
    <div
      class="sc-gKXOVf bQJQqE"
    >
      Mar 20, 2024 10:00 - Mar 20, 2024 12:00
    </div>
    <div
      class="sc-iBkjds bQJQqE"
    >
      <div
        class="sc-jSMfEi bQJQqE"
      >
        <button
          class="sc-furwcr bQJQqE"
        >
          Edit
        </button>
        <button
          class="sc-furwcr bQJQqE"
        >
          Delete
        </button>
      </div>
      <span
        class="sc-jrQzAO bQJQqE"
      >
        Shared
      </span>
    </div>
  </div>
</div>
`;

exports[`ScheduleItem renders schedule description 1`] = `
<div>
  <div
    class="sc-bczRLJ jQJQqE"
  >
    <div
      class="sc-gsnTjR bQJQqE"
    >
      <h3
        class="sc-dkzDqf bQJQqE"
      >
        Test Schedule
      </h3>
      <span
        class="sc-hKMtZM bQJQqE"
        category="WORK"
      >
        Work
      </span>
    </div>
    <p
      class="sc-eCYdqJ bQJQqE"
    >
      This is a test schedule
    </p>
    <div
      class="sc-gKXOVf bQJQqE"
    >
      Mar 20, 2024 10:00 - Mar 20, 2024 12:00
    </div>
    <div
      class="sc-iBkjds bQJQqE"
    >
      <div
        class="sc-jSMfEi bQJQqE"
      >
        <button
          class="sc-furwcr bQJQqE"
        >
          Edit
        </button>
        <button
          class="sc-furwcr bQJQqE"
        >
          Delete
        </button>
      </div>
      <span
        class="sc-jrQzAO bQJQqE"
      >
        Shared
      </span>
    </div>
  </div>
</div>
`;

exports[`ScheduleItem renders schedule time correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ jQJQqE"
  >
    <div
      class="sc-gsnTjR bQJQqE"
    >
      <h3
        class="sc-dkzDqf bQJQqE"
      >
        Test Schedule
      </h3>
      <span
        class="sc-hKMtZM bQJQqE"
        category="WORK"
      >
        Work
      </span>
    </div>
    <p
      class="sc-eCYdqJ bQJQqE"
    >
      This is a test schedule
    </p>
    <div
      class="sc-gKXOVf bQJQqE"
    >
      Mar 20, 2024 10:00 - Mar 20, 2024 12:00
    </div>
    <div
      class="sc-iBkjds bQJQqE"
    >
      <div
        class="sc-jSMfEi bQJQqE"
      >
        <button
          class="sc-furwcr bQJQqE"
        >
          Edit
        </button>
        <button
          class="sc-furwcr bQJQqE"
        >
          Delete
        </button>
      </div>
      <span
        class="sc-jrQzAO bQJQqE"
      >
        Shared
      </span>
    </div>
  </div>
</div>
`;

exports[`ScheduleItem renders shared badge when schedule is shared 1`] = `
<div>
  <div
    class="sc-bczRLJ jQJQqE"
  >
    <div
      class="sc-gsnTjR bQJQqE"
    >
      <h3
        class="sc-dkzDqf bQJQqE"
      >
        Test Schedule
      </h3>
      <span
        class="sc-hKMtZM bQJQqE"
        category="WORK"
      >
        Work
      </span>
    </div>
    <p
      class="sc-eCYdqJ bQJQqE"
    >
      This is a test schedule
    </p>
    <div
      class="sc-gKXOVf bQJQqE"
    >
      Mar 20, 2024 10:00 - Mar 20, 2024 12:00
    </div>
    <div
      class="sc-iBkjds bQJQqE"
    >
      <div
        class="sc-jSMfEi bQJQqE"
      >
        <button
          class="sc-furwcr bQJQqE"
        >
          Edit
        </button>
        <button
          class="sc-furwcr bQJQqE"
        >
          Delete
        </button>
      </div>
      <span
        class="sc-jrQzAO bQJQqE"
      >
        Shared
      </span>
    </div>
  </div>
</div>
`; 